### PR TITLE
Updates to Create Index Examples for azure-search-documents

### PR DESCRIPTION
**Pull request by alzimmermsft**

### CODE REVIEW

File: sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexExample.java [hunk 1]
```diff
@@ -18,23 +19,65 @@
...
             .endpoint(ENDPOINT)
-            .credential(searchApiKeyCredential)
+            .credential(new AzureKeyCredential(ADMIN_KEY))
             .buildClient();
 
-        SearchIndex newIndex = new SearchIndex(INDEX_NAME,
-            Arrays.asList(new SearchField("Name", SearchFieldDataType.STRING).setKey(Boolean.TRUE),
-                new SearchField("Cuisine", SearchFieldDataType.STRING)));
+        // Configure the index using SearchFields
+        String indexName = "hotels";
```
Comment (pallavit) [inline sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexExample.java:31]: From food industry to hospitality? Why the change of heart ;)

File: sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexWithFieldBuilderExample.java [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+        // Cleanup index resource.
+        client.deleteIndex(indexName);
+    }
+
+    /**
+     * A hotel.
+     */
+    public class Hotel {
+        @SimpleField(isKey = true, isFilterable = true, isSortable = true)
```
Comment (pallavit) [inline sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexWithFieldBuilderExample.java:48]: I am assuming the difference between simple and searchable is that one if being indexed and the other is not?

If simple field is not indexed, how is it filterable? Can we search non-indexed fields. If so, I am assuming the performance will be super slow for them?

**Review by pallavit — APPROVED** [review]
:shipit:

File: sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexExample.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexExample.java:31]: Turns out being a chef was too stressful and fast paced for me 😄

File: sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexWithFieldBuilderExample.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/samples/java/com/azure/search/documents/indexes/CreateIndexWithFieldBuilderExample.java:48]: Difference between a `SearchableField` and a `SimpleField` is just that a searchable field is searchable while a simple field isn't
